catroid/src/org/catrobat/catroid/physics/PhysicsLook.java

```diff
@@ -42,15 +45,16 @@ public PhysicsLook(Sprite sprite, PhysicsWorld physicsWorld) {
 		physicsObject = physicsWorld.getPhysicsObject(sprite);
 	}
 
-	@Override
-	public void setVisible(boolean visible) {
-		super.setVisible(visible);
-		updatePhysicsObjectState(true);
-	}
 
 	@Override
 	public void setTransparencyInUserInterfaceDimensionUnit(float percent) {
-		super.setTransparencyInUserInterfaceDimensionUnit(percent);
+		if (percent < 0f) {
+			percent = 0f;
+		} else if (percent >= 100f) {
+			percent = 100f;
+		}
+		alpha = (100f - percent) / 100f;
```

Comment:
I assume you're referring to the code of the overridden method in Look.java
If so, how would you suggest solving this issue - the overriding method may not update the "visible" boolean which the superclass' method does.

I was thinking of implementing another method in Look.java which contains the code in question and takes a boolean as second parameter which is used to determine whether "visible" should be updated. So both methods in question would call this new method passing only a different boolean parameter.

Would this be acceptable? Thought I'd ask before starting a trial-and-error process :)